src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -416,7 +421,7 @@ def _reset_eval_dataloader(
 
         return loader_num_batches, dataloaders
 
-    def _request_dataloader(self, stage: RunningStage) -> Union[DataLoader, List[DataLoader]]:
+    def _request_dataloader(self, stage: RunningStage) -> REQUEST_DATALOADER:
```

Comment:
```suggestion
    def _request_dataloader(self, stage: RunningStage) -> TRAIN_DATALOADERS:
```
from `utilities/types.py`

Reply:
this suggestion causes several errors. the new type was made to address specific errors encountered.